modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/GoClientCodegen.java

```diff
@@ -401,7 +401,13 @@ public String toOperationId(String operationId) {
                 listIterator.add(newImportMap);
             }
         }
-
+        // add pointer to return type string if it is not array
+        for (CodegenOperation operation : operations) {
+            if((operation.returnContainer == null || operation.returnContainer != "array") 
+                && operation.returnType != null) {
+                operation.returnType = "*" + operation.returnType;
+            }
+        }
```

Comment:
@guohuang  Thanks for the PR. What about using the `isListContainer` mustache tag to check if response type is an array?

e.g. `{{#isListContainer}} ... {{/isListContainer}}`

Ref: https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/CodegenResponse.java#L17